### PR TITLE
Fix form field suggestions

### LINKED ISSUE #2826: Problem New custom field in Notification Templates
I have a problem when I try to add a new "custom field" inside a notification templates"

I correctly created a data field categories and added 2 data fields, but when I go to add it to the template notification it gives me an error which I report in the attachment

## Your Environment

* Director version (1.11.0):
* Icinga Web 2 version and modules (2.12.0):
* Icinga 2 version (r2.14.0-1):
* Operating System and version:
* Debian 11.8

Image 003

### PR DESCRIPTION
**Pull request by raviks789**
Other available fields should be suggested when no check command has been chosen for service template.

fixes #2826, #2815

### CODE REVIEW

**Comment by RincewindsHat** [comment]
does not work for notifications for a first test

**Comment by raviks789** [comment]
> does not work for notifications on a first test

As discussed offline, this is a fix for the bug where the data fields are not available for selection when the check commands are not selected (Not a feature to show the custom properties of notification command the notifications form).

**Comment by bigfreak85** [comment]
works and has helped me to get the job done

File: application/forms/IcingaObjectFieldForm.php [hunk 1]
```diff
@@ -167,26 +179,4 @@ public function onSuccess()
         $this->object()->set('var_filter', $this->getValue('var_filter'));
         parent::onSuccess();
     }
-
-    protected static function getFilterFields(IcingaObject $object): array
```
Comment (nilmerg) [inline application/forms/IcingaObjectFieldForm.php:171]: Why did you remove this? It doesn't seem relevant to the fix.

**Review by nilmerg — CHANGES_REQUESTED** [review]

File: test/php/library/Director/Form/IcingaObjectFieldFormTest.php [hunk 2]
```diff
@@ -0,0 +1,153 @@
...
+
+    /** @var ?IcingaCommand */
+    protected $testIcingaCommand = null;
+
+    /** @var IcingaHost */
+    private $testHost;
+
+    public function setUp(): void
+    {
+        $this->app()->setRequest(new Request());
```
Comment (nilmerg) [inline test/php/library/Director/Form/IcingaObjectFieldFormTest.php:35]: This is not required if the director's `BaseTestCase` extends Icinga Web's `BaseTestCase`. At least, works for me locally. Remember to call the parent's `setUp` then.

File: test/php/library/Director/Form/IcingaObjectFieldFormTest.php [hunk 3]
```diff
@@ -0,0 +1,153 @@
...
+
+        if ($this->hasDb()) {
+            $db = $this->getDb();
+            $this->tearDown();
```
Comment (nilmerg) [inline test/php/library/Director/Form/IcingaObjectFieldFormTest.php:39]: Don't call this in `setUp`. I don't even dare to think of the side effects.

File: test/php/library/Director/Form/IcingaObjectFieldFormTest.php [hunk 4]
```diff
@@ -0,0 +1,153 @@
...
+            [
+                'Other available fields' => [
+                    $this->testDatafield->get('id') => "Blah (dataFieldTest)"
+                ]
+            ],
+            $suggestions
+        );
+    }
+
+    public function tearDown(): void
```
Comment (nilmerg) [inline test/php/library/Director/Form/IcingaObjectFieldFormTest.php:125]: Call the parent once `BaseTestCase` extends `Icinga\Test\BaseTestCase`

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — CHANGES_REQUESTED** [review]

**Review by nilmerg — APPROVED** [review]

**Pull request merged**